### ISSUE TITLE
Suggesting Ability, Talent & Item Builds

### ISSUE THREAD

**Issue by Nostrademous** [andreiapostoae/dota2-predictor#13]
Do you think the data mined could be used to train a ML system to make recommendations on the order of ability leveling that should happen in a match given a specific role and lane versus specific opponents?

Similarly, could it make suggestions with respect to which talent to level at each decision point?

Finally, how about a generic Item Build suggestion system based on friendly and enemy hero pool, friendly and enemy current item list, and hero-in-question role?

I'm working on a generic dota2 RL platform and system (modeled after DeepMind's SC2 work) and all 3 of these (plus hero selection real-time during game start) would be useful pieces to have.

My dota 2 work is here: https://github.com/pydota2/pydota2

**Comment by andreiapostoae** [comment]
Hello! The problem with creating a recommender system is not in the algorithm itself, but in the available data. At the moment opendota has a very well made API, but to find information about each game, you would have to make one query for each game at a time. The data mined by me in this project fetches a hundred games at once because I only needed the heroes in each team and the final result.

In conclusion, what you are saying is definitely possible to create with Machine Learning, but it depends a lot on the data you can gather from opendota/Valve official API. 

I'm currently studying reinforcement learning for my diploma (barely started though) and would love to know more about your project as it fits my interests nicely. Send me an e-mail if you would also love a chat

**Comment by Nostrademous** [comment]
I have code to pull down data for one game at a time from dotabuff.com. I will email you to chat separately.

**Comment by andreiapostoae** [comment]
Ha! Valve was ahead of me. :grin:

**Comment by Nostrademous** [comment]
It begs the question what techniques they use.
